documents/oauth_token.md

```diff
@@ -0,0 +1,140 @@
+### OAuthToken
+
+```rb
+require "razorpay"
+```
+
+### Generate Authorize Url
+```rb
+body = {
+    submerchant_id: '<SUBMERCHANT_MID>',
+    timestamp: Time.now.to_i
+}
+onboarding_signature = Razorpay::Utility.generate_onboarding_signature(body, '<YOUR_CLIENT_SECRET>')
+
+options = {
+    'client_id'            => '<YOUR_CLIENT_ID>',
+    'redirect_uri'         => 'https://example.com/razorpay_callback',
+    'scopes'               => ["read_write"],
+    'state'                => 'NOBYtv8r6c75ex6WZ',
+    'onboarding_signature' => onboarding_signature
+}
+authorize_url = Razorpay::OAuthToken.get_auth_url(options)
+```
+
+**Parameters:**
+| Name                 | Type   | Description                                                                                                                                             |
+|----------------------|--------|---------------------------------------------------------------------------------------------------------------------------------------------------------|
+| client_id*           | string | Unique client identifier.                                                                                                                               |
+| redirect_uri*        | string | Callback URL used by Razorpay to redirect after the user approves or denies the authorisation request. The client should whitelist the 'redirect_uri'.  |
+| scopes*              | array  | Defines what access your application is requesting from the user. You can request one or multiple scopes by adding them to an array as indicated above. |
+| state*               | string | A random string generated by your service. This parameter helps prevent cross-site request forgery (CSRF) attacks.                                      |
+| onboarding_signature | string | A cryptographic string generated by your service using generateOnboardingSignature method in Utils class. Only applicable for accounts created with pre-fill KYC |
```

Comment:
Does this formatting needs to be improved?
This is how it looks currently.
<img width="1085" alt="Screenshot 2024-02-28 at 11 48 50 AM" src="https://github.com/razorpay/razorpay-ruby/assets/23275047/a4650774-f114-4b08-92b8-2b8d7e52562d">

Reply:
Formatting was done. Had to add additional line after parameters which was causing this issue.